web.rb

```diff
@@ -61,12 +69,24 @@
   ###
 
   account_id = data['relationships']['account']['data']['id']
-  result = select_account_and_membership(account_id)
-  error('account not found.', 400) if result.empty?
+  result = select_login_data(account_id)
+  error("account not found.", 400) if result.empty?
+
   account_uri = result.first[:account].to_s
   membership_uri = result.first[:membership].to_s
   membership_id = result.first[:membership_id].to_s
 
+  person_status = result.first[:status].to_s
+  organization_status = result.first[:organization_status].to_s
+  membership_status = result.first[:status].to_s
+
+  error("This user is blocked.", 403) if person_status == BLOCKED_STATUS
+  if organization_status == BLOCKED_STATUS
+    insert_membership_block(membership_uri)
+    error("This organization is blocked.", 403) if organization_status == BLOCKED_STATUS
```

Comment:
`if ...` is not required since we're already inside a wrapping `if`-block with the same check